PWGHF/D2H/TableProducer/candidateCreatorDV0Reduced.cxx

```diff
@@ -0,0 +1,254 @@
+      return false;
+    }
+    return true;
+  }
+
+  template <DecayChannel channel, typename DRedTable, typename V0RedTable>
+  bool isV0Selected(V0RedTable const& candV0, DRedTable const& candD)
+  {
+    float massV0{0.};
+    float invMassV0{0.};
+    // slection on V0 candidate mass
+    if (channel == DecayChannel::Dstar2ToDplusK0s || channel == DecayChannel::Ds1ToDstarK0s) {
+      massV0 = massK0;
+      invMassV0 = candV0.invMassK0s();
+    } else if (channel == DecayChannel::XcToDplusLambda) {
+      massV0 = massLambda;
+      if (candD.dType() > 0) {
+        invMassV0 = candV0.invMassLambda();
+      } else {
+        invMassV0 = candV0.invMassAntiLambda();
+      }
+    }
+    if (std::fabs(invMassV0 - massV0) > invMassWindowV0) {
+      return false;
+    }
+    return true;
+  }
+
+  template <DecayChannel channel, typename Coll, typename DRedTable, typename V0RedTable>
+  void runCandidateCreation(Coll const& collisions,
+                            DRedTable const& candsD,
+                            V0RedTable const& candsV0)
+  {
+    // loop on D candidates
+    for (const auto& candD : candsD) {
+      // selection of D candidates
+      if (activateQA) {
+        registry.fill(HIST("hSelections"), 1);
+      }
+      if (!isDSelected<channel>(candD)) {
+        continue;
+      }
+      if (activateQA) {
+        registry.fill(HIST("hSelections"), 1 + Selections::DSel);
+      }
+      float invMassD = candD.invMass();
+      std::array<float, 3> pVecD = {candD.px(), candD.py(), candD.pz()};
+      float ptD = RecoDecay::pt(pVecD);
+      ;
+      // loop on V0 candidates
+      bool already_counted{false};
+      for (const auto& candV0 : candsV0) {
+        if (!isV0Selected<channel>(candV0, candD)) {
+          continue;
+        }
+        if (activateQA && !already_counted) {
+          registry.fill(HIST("hSelections"), 1 + Selections::V0Sel);
+          already_counted = true;
+        }
+        float invMass2Reso{0.};
+        float invMassV0{0.};
+        std::array<float, 3> pVecV0 = {candV0.px(), candV0.py(), candV0.pz()};
+        float ptV0 = RecoDecay::pt(pVecV0);
+        float ptReso = RecoDecay::pt(RecoDecay::sumOfVec(pVecV0, pVecD));
+        switch (channel) {
+          case DecayChannel::Ds1ToDstarK0s:
+            invMassV0 = candV0.invMassK0s();
+            invMass2Reso = RecoDecay::m2(std::array{pVecD, pVecV0}, std::array{massDstar, massK0});
+            registry.fill(HIST("hMassDs1"), std::sqrt(invMass2Reso), ptReso);
+            break;
+          case DecayChannel::Dstar2ToDplusK0s:
+            invMassV0 = candV0.invMassK0s();
+            invMass2Reso = RecoDecay::m2(std::array{pVecD, pVecV0}, std::array{massDplus, massK0});
+            registry.fill(HIST("hMassDsStar2"), std::sqrt(invMass2Reso), ptReso);
+            break;
+          case DecayChannel::XcToDplusLambda:
+            if (candD.dType() > 0) {
+              invMassV0 = candV0.invMassLambda();
```

Comment:
Here you should also check that the V0 is compatible with the right hypothesis (`Lambda` in case of `D+` and `AntiLambda` in case of `D-`). In the future it would be good to add an option to do also wrong sign (but then it should be flagged in the output)

Reply:
Thanks! You are right, I forgot it